www/js/app.js

```diff
@@ -677,7 +677,6 @@ define(['jquery', 'zimArchiveLoader', 'uiUtil', 'settingsStore','abstractFilesys
                 // If this is the first time we are initiating the SW, allow Promises to complete by delaying potential reload till next tick
                 delay = 0;
             }
-            messageChannel = tmpMessageChannel;
```

Comment:
I'm not sure I understand why this has been removed. Is it because we don't actually need to store this message channel?

Reply:
In fact, the global variable `messageChannel` was defined, assigned here, but never used.
I suppose we were keeping a single messageChannel instance at some point in the past, and afterwards we changed our mind.
So I removed the variable declaration, and assignment here.

Reply:
OK. At some point (not now) it might be good to check whether the implementation still needs the Service Worker to be kept alive. The Service Worker spec is specifically designed to put the SW to sleep when not being used, and it should fire up again immediately it is invoked again. It's possible, in fact, that an older implementation tried to keep the messageChannel alive (hence we have a global messageChannel assignment here), which should not be necessary. Each use of the SW can define a new messageChannel. I'm only speculating here, nothing tested!

Reply:
I doubt we could simply remove this keepalive implementation.
The SW can be removed and re-created by the browser, by design from the beginning.
The problem is that, when it's re-created, it does not have the messageChannel variable to ask app.js for the content. We send this messageChannel through the "init" message.

When I faced this issue in https://github.com/kiwix/kiwix-js/issues/145, I did not find a better solution.
But I'd be very glad to have a better one.

Reply:
OK, we can investigate another time. Fine to merge this if you've done some testing on SW mode.